tests/platform_tests/mellanox/test_thermal_control.py

```diff
@@ -65,64 +75,56 @@ def test_set_psu_fan_speed(duthosts, rand_one_dut_hostname, mocker_factory):
+
+    logger.info('Mock FAN presence...')
     single_fan_mocker.mock_presence()
-    wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, check_cooling_cur_state, duthost, MAX_COOLING_LEVEL, operator.ne)
-    logging.info('Wait {} seconds for the policy to take effect...'.format(THERMAL_CONTROL_TEST_WAIT_TIME * 2))
-    # We have to wait THERMAL_CONTROL_TEST_WAIT_TIME * 2 seconds long here because:
-    #     Usually we only need wait THERMAL_CONTROL_TEST_WAIT_TIME seconds here to make sure thermal
-    #     control daemon change the cooling level to proper value, However,
-    #     there is chance that kernel might change cooling state back to MAX_COOLING_LEVEL after
-    #     user space thermal control adjust it to dynamic minimum value. So we have to wait longer for the
-    #     user space thermal control to set fan speed to dynamic minimum value again. It
-    #     means that we might need wait up to 2 thermal loops here.
-    time.sleep(THERMAL_CONTROL_TEST_WAIT_TIME * 2)
-    cooling_cur_state = get_cooling_cur_state(duthost)
-    if cooling_cur_state == MAX_COOLING_LEVEL:
-        cmd_output = str(duthost.command('show platform temperature')['stdout_lines'])
-        cmd_output = cmd_output.replace("u'", "").replace(',', " ")
-        cmd_output = re.split(r'  +',cmd_output)
-        cmd_output.pop(0)
-        j = 0
-        table = []
-        while j != len(cmd_output):
-            entry = []
-            for i in range(8):
-                entry.append(cmd_output[j + i])
-            table.append(entry)
-            j += 8
-        pytest.skip('Cooling level is still 10, ignore the rest test.\nIt might happen because the asic temperature is still high.\nCurrent system temperature:\n{}'.format(tabulate(table)))
-    logging.info('Cooling level changed to {}'.format(cooling_cur_state))
-    if cooling_cur_state < 6: # PSU fan speed will never be less than 60%
-        cooling_cur_state = 6
-    for index in range(psu_num):
-        speed = get_psu_speed(duthost, index)
-        logging.info('Speed for PSU {} fan is {}'.format(index, speed))
-        _check_psu_fan_speed_in_range(speed, psu_max_speed, cooling_cur_state)
+    wait_result = wait_until(THERMAL_CONTROL_TEST_WAIT_TIME * 2, 
+                             THERMAL_CONTROL_TEST_CHECK_INTERVAL, 
+                             check_psu_fan_speed, 
+                             duthost, 
+                             psu_num, 
+                             psu_max_speed, 
+                             operator.ne)
+
+    if not wait_result:
+        cooling_cur_state = get_cooling_cur_state(duthost)
+        if cooling_cur_state == MAX_COOLING_LEVEL:
+            cmd_output = str(duthost.command('show platform temperature')['stdout_lines'])
+            cmd_output = cmd_output.replace("u'", "").replace(',', " ")
+            cmd_output = re.split(r'  +',cmd_output)
+            cmd_output.pop(0)
```

Comment:
We need to add a `--json` flag to the `show platform temperature` command (and many other commands), so that we can more easily get the proper fields in instances like this.

Reply:
Agree.